Raven.Database/Bundles/Replication/Controllers/ReplicationController.cs

```diff
@@ -562,7 +578,14 @@ private void SaveReplicationSource(string src, string lastEtag, int batchSize, s
                                  new RavenJObject(), null);
                 });
             }
-
+            //We don't want to delay large batches of attachments otherwise we will endup with alot of data that needs to get replicated.
+            if (Request.Content.Headers.ContentLength > Database.Configuration.Replication.ReplicationPropagationDelaySizeInBytes)
+            {
```

Comment:
Wrong here, you are checking just the _single_ batch, what about all pending batches?
And why do we care about the size in the first place?
We are already doing size limiting when we are replication to destinations